### PR TITLE
Remove bounds on T for Debug, PartialEq and Hash impls

### LINKED ISSUE #12: Remove bounds on `T` for `Debug`, `PartialEq` and `Hash` impls
`RawParts` derives `Debug`, `PartialEq`, and `Hash`. The derive macros from `core` add bounds on `T` even when they are not required, as is the case with the `T` in `RawParts` which is only a pointer and impls these traits regardless of what `T` is.

Remove the derived impls and manually implement these traits.

Such a change warrants a minor version bump.

### PR DESCRIPTION
**Pull request by molinarim**
Resolves #12. Made suggested changes from #16. Reorganized imports, added impl Eq, added vim to .gitignore.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -311,4 +339,64 @@ mod tests {
...
+        assert_ne!(raw_parts_1, raw_parts_2);
+    }
+
+    #[test]
+    fn partial_eq_pass() {
+        let mut vec = Vec::with_capacity(100); // capacity is 100
+        vec.extend_from_slice(b"123456789"); // length is 9
+
+        let raw_parts = RawParts::from_vec(vec);
+        assert_eq!(raw_parts, raw_parts);
```
Comment (lopopolo) [inline src/lib.rs]: We can address this clippy warning with the suggestion below:

```
warning: identical args used in this `assert_eq!` macro call
   --> src/lib.rs:400:20
    |
400 |         assert_eq!(raw_parts, raw_parts);
    |                    ^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:1:9
    |
1   | #![warn(clippy::all)]
    |         ^^^^^^^^^^^
    = note: `#[warn(clippy::eq_op)]` implied by `#[warn(clippy::all)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eq_op
```

```suggestion
        let RawParts {
            ptr,
            length,
            capacity,
        } = RawParts::from_vec(vec);
        let a = RawParts {
            ptr,
            length,
            capacity,
        };
        let b = RawParts {
            ptr,
            length,
            capacity,
        };
        assert_eq!(a, b);
```

**Review by lopopolo — APPROVED** [review]

**Comment by lopopolo** [comment]
🎉 congrats on getting your first GitHub PR merged @molinarim!

I'm not sure if you know this, but next time you can just commit your changes I requested on the same branch and do another `git push`, which would have updated #16 without having to open another PR.

Thanks for contributing. I'll work on prepping this change for the v1.1.0 release.

File: src/lib.rs (hunk 1 above)
Comment (molinarim) [reply src/lib.rs]: Thanks for fixing that, didn't realize it would trigger a warning.

**Comment by molinarim** [comment]
Thank you!

It was nice to get back into Rust. My favorite topics in school were on language implementation, so I enjoyed working on this and I'll keep an eye out for any other issues I might want to tackle.

I'll go ahead and squash like you mentioned.

**Pull request merged**